### DISCUSSION TITLE
Multiple chains are not supported after migrated to v2

### DISCUSSION THREAD

**Discussion by reevancutinho** [WalletConnect/walletconnect-monorepo#3175] [Developer Support]
Hi, 
I was using @walletconnect/react-native-dapp before and now it is discontinued so i upgraded to @walletconnect/modal-react-native. Now whenever i add single chain in sessionParams it works fine, but when there are multiple chains in sessionParams it doesnt give a popup in web3 wallet to connect. Heres the below code:

```
`export const sessionParams = {
  namespaces: {
    eip155: {
      chains: [
        "eip155:1",   // Ethereum
        "eip155:137"  // Polygon
        ],
      "methods": ["eth_sign"],
      events: ["accountsChanged"],
      rpcMap: {},
    },
  },
};`
```

Any help would be appreciated. i just want to retrieve wallet address and chain id after connecting.
or is there any way i can identify the type of chain (ethereum or polygon) from the provider information we get after connecting?

**Comment by glitch-txs** [comment]
Hi, which wallet are you testing with? You can alternatively add optionaNamespaces object that can be accepted by the wallet or not without interrupting the conenction.

**Reply by reevancutinho** [reply]
Hi i tested with metamask and zerion wallets.  I tried adding optionalNamespaces as well but its still the same

**Reply by glitch-txs** [reply]
Can you try switching eth_sign for personal_sign? This one is not supported by most wallets

**Comment by MindlessSteel** [comment]
I have just went threw a issue
Just like this, my crypto.com
Defi wallet was showing up in
The wallet connect protocol
And now it's not? I have just set
Up a account with this app and 
Platform and was trying too contact support but can't
Set too Github instead?
I have unstoppable domains that I still need too mint. Never mind created a dapp or a project for this protocol. Sorry about this not being a solution🙈👀
